nixos/modules/services/desktops/pipewire/pipewire-media-session.nix

```diff
@@ -0,0 +1,336 @@
+                # this matches all cards
+                device.name = "~alsa_card.*";
+              }
+            ];
+            actions = {
+              # actions can update properties on the matched object.
+              update-props = {
+                api.alsa.use-acp = true;
+                #api.alsa.use-ucm = true
+                #api.alsa.soft-mixer = false
+                #api.alsa.ignore-dB = false
+                #device.profile-set = "profileset-name"
+                #device.profile = "default profile name"
+                api.acp.auto-profile = false;
+                api.acp.auto-port = false;
+                #device.nick = "My Device"
+              };
+            };
+          }
+          {
+            matches = [
+              {
+                # matches all sinks
+                node.name = "~alsa_input.*";
+              }
+              {
+                # matches all sources
+                node.name = "~alsa_output.*";
+              }
+            ];
+            actions = {
+              update-props = {
+                #node.nick = 			"My Node"
+                #node.nick = 			null
+                #priority.driver = 		100
+                #priority.session = 		100
+                #node.pause-on-idle = 		false
+                #resample.quality = 		4
+                #channelmix.normalize =		false
+                #channelmix.mix-lfe = 		false
+                #audio.channels = 		2
+                #audio.format = 		"S16LE"
+                #audio.rate = 			44100
+                #audio.position = 		"FL,FR"
+                #api.alsa.period-size =         1024
+                #api.alsa.headroom =            0
+                #api.alsa.disable-mmap =        false
+                #api.alsa.disable-batch =       false
+              };
+            };
+          }
+          ];
+        };
+      };
+
+      bluezMonitorConfig = mkOption {
+        type = types.attrs;
+        description = ''
+          Configuration for the bluez5 monitor.
+        '';
+        default = {
+          # bluez-monitor config file
+          properties = {
+            # msbc is not expected to work on all headset + adapter combinations.
+            #bluez5.msbc-support = true
+            #bluez5.sbc-xq-support = true
+
+            # Enabled headset roles (default: [ hsp_hs hfp_ag ]), this
+            # property only applies to native backend. Currently some headsets
+            # (Sony WH-1000XM3) are not working with both hsp_ag and hfp_ag
+            # enabled, disable either hsp_ag or hfp_ag to work around it.
+            #
+            # Supported headset roles: hsp_hs (HSP Headset),
+            #                          hsp_ag (HSP Audio Gateway),
+            #                          hfp_ag (HFP Audio Gateway)
+            #bluez5.headset-roles = [ hsp_hs hsp_ag hfp_ag ]
+
+            # Enabled A2DP codecs (default: all)
+            #bluez5.codecs = [ sbc aac ldac aptx aptx_hd ]
```

Comment:
Should the commented out code not be at least valid nix code?

Reply:
Certainly wouldn't hurt. On the other hand the whole thing should be redone and probably autogenerated from upstream config files. Whenever someone has the time and knowledge to implement that.